_drafts/zio-fibers.md

```diff
@@ -0,0 +1,290 @@
+
+```shell
+[zio-default-async-2]: Going to the bathroom
+[zio-default-async-1]: Boiling some water
+```
+
+As we expect, ZIO executed the two effects concurrently on different threads. Just to remember, concurrently running a set of tasks means that the order in which the runtime performs them is irrelevant.
+
+### 6. Synchronizing with a Fiber
+
+The astute reader should have noticed that boiling the water without preparing a good coffee is non-sense. However, Bob can't prepare the coffee without the boiled water. Moreover, Bob can't prepare the coffee if he's still in the bathroom. So, we need to synchronize the action of preparing coffee to the results of the previous two activities.
+
+ZIO fibers provide the `join` method to wait for the termination of a fiber:
+
+```scala
+// From ZIO library
+trait Fiber[+E, +A] {
+  def join: IO[E, A]
+}
+```
+
+Through the `join` method, we can wait for the result of concurrent computation and eventually use it:
+
+```scala
+def concurrentWakeUpRoutine(): ZIO[Any, Nothing, Unit] = for {
+  bathFiber <- bathTime.debug(printThread).fork
+  boilingFiber <- boilingWater.debug(printThread).fork
+  zippedFiber = bathFiber.zip(boilingFiber)
+  result <- zippedFiber.join.debug(printThread)
+  _ <- ZIO.succeed(s"$result...done").debug(printThread) *> preparingCoffee.debug(printThread)
+} yield ()
+```
+
+However, in our example, we need to wait for the completion of two concurrent fibers. So, we need to combine them first and then join the resulting fiber. Thus, the `zip` method combines two fibers into a single fiber that produces both results.
+
+Joining a fiber lets us also gather the result of its execution. In the example, the variable `result` has type `(String, String)`, and we successfully use it in the next step of the routine. We also introduced the `*>` operator, which is an alias for the `zipRight` function, and let us concatenate the execution of two effects not depending on each other.
+
+The execution of the `concurrentWakeUpRoutine` function prints precisely what we expect:
+
+```shell
+[zio-default-async-2]: Going to the bathroom
+[zio-default-async-3]: Boiling some water
+[zio-default-async-6]: (Going to the bathroom,Boiling some water)
+[zio-default-async-6]: (Going to the bathroom,Boiling some water)...done
+[zio-default-async-6]: Preparing the coffee
+```
+
+Bob uses the fiber to go to the bathroom, and the fiber that boils the water runs concurrently on different threads. Then, ZIO executes the fiber used to prepare the coffee in a new thread only after the previous fibers succeeded.
+
+However, what is happening is different from using systems threads directly to represent concurrent computation. In fact, **ZIO fibers don't block any thread during the waiting** associated with the call of the `join` method. Just remember that a `Fiber` represents only a data structure, a blueprint of computation, and not the computation itself.
```

Comment:
"However, what is happening is different from using systems threads directly to represent concurrent computation" -> "However, this code is not using systems threads directly for concurrent computation"

"In fact" -> "In fact, the ZIO runtime is smart to free up threads when they're not active, so"